### DISCUSSION TITLE
HIP-794: Sunsetting Balance File

### DISCUSSION THREAD

**Discussion by steven-sheehy** [hiero-ledger/hiero-improvement-proposals#798] [Edits]
Thread to discuss [HIP-794](https://hips.hedera.com/hip/hip-794).

**Comment by david-bakin-sl** [comment]
w.r.t. "open issue":
> It is under discussion if balance files can be provided with a process that does not involve analyzing the data on consensus nodes.

You should be able to produce them from signed state files.
* Since these are produced every 15 minutes on (every) consensus node and uploaded to a cloud bucket there'd be some latency (which would vary up to, ~20min, I imagine, though someone else might have better numbers on how long it takes a signed state file from some given 15-minute clock time to get to the cloud, some time to download them, and some time to load the signed state file into a tool.
* (For what it's worth, on my Macbook Pro (2021, Apple M1 Max processor) given a signed state file from mainnet, 2021-07-19, it takes:
  * 35s to read the signed state file into memory
  * <1s to iterate over the 2.99M accounts collecting _all_ information from them
  * 30s to _format_ and write 550MB of the _entire contents_ of accounts (i.e., not just the balance)
  So the latency would be the 15m + cloud upload then download + ~1m more to process it. The tool used is the signed state dumper (accounts subcommand) found in `hedera-node/cli-clients`.)
